rcl/src/rcl/logging_rosout.c

```diff
@@ -292,6 +343,122 @@ void rcl_logging_rosout_output_handler(
   }
 }
 
+static char *
+_rcl_logging_rosout_get_sublogger_name(
+  const char * logger_name, const char * sublogger_name)
+{
+  char * full_sublogger_name = NULL;
+
+  full_sublogger_name = rcutils_format_string(
+    __rosout_allocator, "%s%s%s",
+    logger_name, RCUTILS_LOGGING_SEPARATOR_STRING, sublogger_name);
+  if (NULL == full_sublogger_name) {
+    RCL_SET_ERROR_MSG("Failed to allocate a full sublogger name.");
+  }
+
+  return full_sublogger_name;
+}
+
+rcl_ret_t
+rcl_logging_rosout_add_sublogger(
+  const char * logger_name, const char * sublogger_name)
+{
+  rcl_ret_t status = RCL_RET_OK;
+  char * full_sublogger_name = NULL;
+  rosout_map_entry_t entry;
+
+  RCL_LOGGING_ROSOUT_VERIFY_INITIALIZED
+  if (NULL == logger_name) {
+    RCL_SET_ERROR_MSG("Logger name was null.");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  if (NULL == sublogger_name) {
+    RCL_SET_ERROR_MSG("Sub-logger name was null.");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
```

Comment:
these are redundant for add/remove API, this could be checked in `_rcl_logging_rosout_get_sublogger_name`?

Reply:
OK, I'll update it because we could call https://github.com/ros2/rcl/blob/108d4fa793d69933d8d8f298f6f868c5dbb2f8ba/rcl/src/rcl/logging_rosout.c#L380
after getting the `full_sublogger_name`.